test/passes/optimize-instructions_all-features.wast

```diff
@@ -6162,3 +6162,10 @@
     )
   )
 )
+;; typed function references
+(module
+  (type $i32-i32 (func (param i32) (result i32)))
+  (func $call_from-param (param $f (ref null $i32-i32)) (result i32)
+    (unreachable)
+  )
+)
```

Comment:
This is not supposed to be optimized, right? It would be good to comment on what the expected output is and why it shouldn't be changed.

Reply:
Sure, adding a comment.

Reply:
Sorry, this is maybe because I'm not very famliar with `OptimizeInstructions`, but I think it's worth a little more explanation because that pass is doing so many things anyway. If this has not been a ref type, what kind of optimization would have happened? I'm still not very sure what this tries to test....

Reply:
Oh, no optimization is possible there. The bug was just a very silly one. We always scan the parameters. We just need to not crash while doing that, if the type is a ref type. I'll add more to the comment, sorry for not being clear.